tests/generate_status_reference_data.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python
+# Copyright (c) 2018 The Regents of the University of Michigan
+# All rights reserved.
+# This software is licensed under the BSD 3-Clause License.
+import argparse
+import os
+
+import signac
+from define_status_test_project import _TestProject
+from test_project import redirect_stdout
+
+import flow
+import flow.environments
+from flow.scheduling.fake_scheduler import FakeScheduler
+
+PROJECT_NAME = "StatusTest"
+ARCHIVE_DIR = os.path.normpath(
+    os.path.join(os.path.dirname(__file__), "./status_reference_data.tar.gz")
+)
+
+
+def init(project):
+    """Initialize the data space for the given project."""
+
+    for a in range(2, 4):
+        for b in range(2):
+            project.open_job(dict(a=a, b=b)).init()
+            project.open_job(dict(a=dict(a=a), b=b)).init()
+
+
+def init_status_options(project):
+
+    options = [{}, {"detailed": True}, {"parameters": ["a"]}, {"expand": True}]
```

Comment:
Add more options here. I would recommend first checking their outputs and making sure that you understand the difference that each option makes. List of options: https://docs.signac.io/projects/flow/en/latest/api.html#flow.FlowProject.print_status

Here is a suggested list of options from the Python and CLI options. I would check on the "max lines/width" options to see if the values I suggested are reasonable or not. We might need to tweak the FlowProject or state points to properly test some of these options.
```suggestion
    options = [
        {},  # default options
        {"overview": False},  # --no-overview
        {"overview_max_lines": 2},  # --overview-max-lines 2
        {"detailed": True},  # -d, --detailed
        {"parameters": ["a"]},  # -p a, --parameters a
        {"param_max_width": 1},  # --param-max-width 1
        {"expand": True},  # -e, --expand
        {"all_ops": True},  # -a, --all-operations
        {"detailed": True, "all_ops": True},  # --full
        {"only_incomplete": True},  # --only-incomplete-operations
        {"dump_json": True},  # --json
        {"unroll": False},  # --stack
        {"compact": True},  # -1, --one-line
        {"pretty": True},  # --pretty
        {"profile": True},  # No CLI flag
        {"eligible_jobs_max_lines": 2},  # --eligible-jobs-max-lines 2
        {"output_format": "markdown"},  # -o markdown, --output-format markdown
        {"output_format": "html"},  # -o html, --output-format html
    ]
```

Reply:
I've added all the options. I have a question about `{'profile': True}` - could you explain what this does? The script `generate_status_reference_data.py` fails when I add this option.

The error I got:
```
Traceback (most recent call last):
  File "/Users/zshiqi/code/signac-flow/tests/generate_status_reference_data.py", line 103, in <module>
    main(parser.parse_args())
  File "/Users/zshiqi/code/signac-flow/tests/generate_status_reference_data.py", line 88, in main
    fp.print_status(**kwargs, file=status_file)
  File "/Users/zshiqi/code/signac-flow/flow/project.py", line 2684, in print_status
    if re.match(profile, module_fn):
  File "/Users/zshiqi/miniconda3/envs/signac-flow-dev/lib/python3.9/re.py", line 191, in match
    return _compile(pattern, flags).match(string)
  File "/Users/zshiqi/miniconda3/envs/signac-flow-dev/lib/python3.9/re.py", line 303, in _compile
    raise TypeError("first argument must be string or compiled pattern")
TypeError: first argument must be string or compiled pattern
```

Reply:
@Charlottez112 From what I saw on Slack, it sounded like you might have discussed the `profile` option with @atravitz during office hours. If there are a small number of options like `profile` that are problematic, we can just skip them in the tests.